src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,157 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (isEmpty()) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Node<T> node = new Node<>(null, value, null);
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index - " + index
+                    + " out of bounds for length: " + size);
```

Comment:
You already have `checkSize` method, no need to duplicate code. Try to add new parameter `bound` to your `checkIndex` and pass `size + 1` here and `size` in other cases.

Reply:
oooo! its GREAT solution!!! thx